flintrock/flintrock.py

```diff
@@ -297,6 +305,21 @@ def launch(
         raise UnsupportedProviderError(provider)
 
 
+def get_last_commit_sha(git_repository: str):
+    """
+    Retrieve the last commit's SHA from a repository hosted on Github.
+    """
+    parsed_url = urllib.parse.urlparse(git_repository)
+    if parsed_url.netloc != "github.com":
+        raise UsageError('Error: Retrieving the last commit\'s SHA only works with '
+                         'repositories hosted on Github.')
+    owner_repo = parsed_url.path[:-4] if parsed_url.path.endswith(".git") else parsed_url.path
```

Comment:
This is a nice touch.

Actually, it makes me wonder: Does everything work if we drop the `.git` from the default repo name? I noticed the config template includes it, and I forget if that was a necessity. If not, it's better to remove it.